core/src/components/datetime/datetime.tsx

```diff
@@ -2383,35 +2383,24 @@ export class Datetime implements ComponentInterface {
                     if (isAdjacentDay) {
                       // The user selected a day outside the current month. Ignore this button, as the month will be re-rendered.
                       this.el.blur();
-                    }
-
-                    this.setWorkingParts({
-                      ...this.workingParts,
-                      month: _month,
-                      day,
-                      year: _year,
-                      isAdjacentDay,
-                    });
-
-                    // multiple only needs date info, so we can wipe out other fields like time
-                    if (multiple) {
-                      this.setActiveParts(
-                        {
-                          month: _month,
-                          day,
-                          year: _year,
-                          isAdjacentDay,
-                        },
-                        isActive
-                      );
+                      this.activeParts = { ...activePart, ...referenceParts };
+                      this.animateToDate(referenceParts);
+                      this.confirm();
                     } else {
-                      this.setActiveParts({
-                        ...activePart,
-                        month: _month,
-                        day,
-                        year: _year,
-                        isAdjacentDay,
+                      this.setWorkingParts({
+                        ...this.workingParts,
+                        ...referenceParts,
                       });
+
+                      // multiple only needs date info, so we can wipe out other fields like time
```

Comment:
Let's make sure to use proper grammar.

```suggestion
                      // Multiple only needs date info so we can wipe out other fields like time.
```